{"level":"error","message":{"code":"ERR_HTTP_HEADERS_SENT"}}
{"level":"error","message":"Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client\n    at ServerResponse.setHeader (_http_outgoing.js:485:11)\n    at ServerResponse.header (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:767:10)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:170:12)\n    at ServerResponse.json (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:267:15)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:158:21)\n    at C:\\Users\\diogo\\API\\controllers\\user.controller.js:84:32\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:246:48\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:135:16\n    at processTicksAndRejections (internal/process/task_queues.js:75:11)"}
{"level":"error","message":{"code":"ERR_HTTP_HEADERS_SENT"}}
{"level":"error","message":"Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client\n    at ServerResponse.setHeader (_http_outgoing.js:485:11)\n    at ServerResponse.header (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:767:10)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:170:12)\n    at ServerResponse.json (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:267:15)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:158:21)\n    at C:\\Users\\diogo\\API\\controllers\\user.controller.js:84:32\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:246:48\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:135:16\n    at processTicksAndRejections (internal/process/task_queues.js:75:11)"}
{"level":"error","message":{}}
{"level":"error","message":"TypeError: app.set is not a function\n    at Object.<anonymous> (C:\\Users\\diogo\\API\\bin\\www:16:5)\n    at Module._compile (internal/modules/cjs/loader.js:959:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)\n    at Module.load (internal/modules/cjs/loader.js:815:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:727:14)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:1047:10)\n    at internal/main/run_main_module.js:17:11"}
{"level":"error","message":{}}
{"level":"error","message":"TypeError: Cannot read property 'set' of undefined\n    at Object.<anonymous> (C:\\Users\\diogo\\API\\bin\\www:16:5)\n    at Module._compile (internal/modules/cjs/loader.js:959:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)\n    at Module.load (internal/modules/cjs/loader.js:815:32)\n    at Function.Module._load (internal/modules/cjs/loader.js:727:14)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:1047:10)\n    at internal/main/run_main_module.js:17:11"}
{"level":"error","message":{"code":"ERR_HTTP_HEADERS_SENT"}}
{"level":"error","message":"Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client\n    at ServerResponse.setHeader (_http_outgoing.js:485:11)\n    at ServerResponse.header (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:767:10)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:170:12)\n    at ServerResponse.json (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:267:15)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:158:21)\n    at C:\\Users\\diogo\\API\\controllers\\user.controller.js:84:32\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:246:48\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:135:16\n    at processTicksAndRejections (internal/process/task_queues.js:75:11)"}
{"level":"error","message":{"code":"ERR_HTTP_HEADERS_SENT"}}
{"level":"error","message":"Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client\n    at ServerResponse.setHeader (_http_outgoing.js:485:11)\n    at ServerResponse.header (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:767:10)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:170:12)\n    at ServerResponse.json (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:267:15)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:158:21)\n    at C:\\Users\\diogo\\API\\controllers\\user.controller.js:84:32\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:246:48\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:135:16\n    at processTicksAndRejections (internal/process/task_queues.js:75:11)"}
{"level":"error","message":{"code":"ERR_HTTP_HEADERS_SENT"}}
{"level":"error","message":"Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client\n    at ServerResponse.setHeader (_http_outgoing.js:485:11)\n    at ServerResponse.header (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:767:10)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:170:12)\n    at ServerResponse.json (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:267:15)\n    at ServerResponse.send (C:\\Users\\diogo\\API\\node_modules\\express\\lib\\response.js:158:21)\n    at C:\\Users\\diogo\\API\\controllers\\user.controller.js:85:32\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\mongoose\\lib\\model.js:4609:16\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:246:48\n    at C:\\Users\\diogo\\API\\node_modules\\kareem\\index.js:135:16\n    at processTicksAndRejections (internal/process/task_queues.js:75:11)"}
{"level":"error","message":{"code":"ERR_STREAM_DESTROYED","source":"socket"}}
{"level":"error","message":"Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed\n    at doWrite (_stream_writable.js:427:19)\n    at clearBuffer (_stream_writable.js:551:7)\n    at Socket.Writable.uncork (_stream_writable.js:325:7)\n    at JSStreamSocket.doWrite (internal/js_stream_socket.js:162:17)\n    at JSStream.onwrite (internal/js_stream_socket.js:20:57)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"DEPTH_ZERO_SELF_SIGNED_CERT","source":"socket"}}
{"level":"error","message":"Error: self signed certificate\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)\n    at TLSSocket.emit (events.js:210:5)\n    at TLSSocket._finishInit (_tls_wrap.js:794:8)\n    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"code":"ERR_STREAM_DESTROYED","source":"socket"}}
{"level":"error","message":"Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed\n    at doWrite (_stream_writable.js:427:19)\n    at clearBuffer (_stream_writable.js:551:7)\n    at Socket.Writable.uncork (_stream_writable.js:325:7)\n    at JSStreamSocket.doWrite (internal/js_stream_socket.js:162:17)\n    at JSStream.onwrite (internal/js_stream_socket.js:20:57)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"DEPTH_ZERO_SELF_SIGNED_CERT","source":"socket"}}
{"level":"error","message":"Error: self signed certificate\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)\n    at TLSSocket.emit (events.js:210:5)\n    at TLSSocket._finishInit (_tls_wrap.js:794:8)\n    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"ERR_STREAM_DESTROYED","source":"socket"}}
{"level":"error","message":"Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed\n    at doWrite (_stream_writable.js:427:19)\n    at clearBuffer (_stream_writable.js:551:7)\n    at Socket.Writable.uncork (_stream_writable.js:325:7)\n    at JSStreamSocket.doWrite (internal/js_stream_socket.js:162:17)\n    at JSStream.onwrite (internal/js_stream_socket.js:20:57)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"DEPTH_ZERO_SELF_SIGNED_CERT","source":"socket"}}
{"level":"error","message":"Error: self signed certificate\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)\n    at TLSSocket.emit (events.js:210:5)\n    at TLSSocket._finishInit (_tls_wrap.js:794:8)\n    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"ERR_STREAM_DESTROYED","source":"socket"}}
{"level":"error","message":"Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed\n    at doWrite (_stream_writable.js:427:19)\n    at clearBuffer (_stream_writable.js:551:7)\n    at Socket.Writable.uncork (_stream_writable.js:325:7)\n    at JSStreamSocket.doWrite (internal/js_stream_socket.js:162:17)\n    at JSStream.onwrite (internal/js_stream_socket.js:20:57)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"DEPTH_ZERO_SELF_SIGNED_CERT","source":"socket"}}
{"level":"error","message":"Error: self signed certificate\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)\n    at TLSSocket.emit (events.js:210:5)\n    at TLSSocket._finishInit (_tls_wrap.js:794:8)\n    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"ERR_STREAM_DESTROYED","source":"socket"}}
{"level":"error","message":"Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed\n    at doWrite (_stream_writable.js:427:19)\n    at clearBuffer (_stream_writable.js:551:7)\n    at Socket.Writable.uncork (_stream_writable.js:325:7)\n    at JSStreamSocket.doWrite (internal/js_stream_socket.js:162:17)\n    at JSStream.onwrite (internal/js_stream_socket.js:20:57)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"DEPTH_ZERO_SELF_SIGNED_CERT","source":"socket"}}
{"level":"error","message":"Error: self signed certificate\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)\n    at TLSSocket.emit (events.js:210:5)\n    at TLSSocket._finishInit (_tls_wrap.js:794:8)\n    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"ERR_STREAM_DESTROYED","source":"socket"}}
{"level":"error","message":"Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed\n    at doWrite (_stream_writable.js:427:19)\n    at clearBuffer (_stream_writable.js:551:7)\n    at Socket.Writable.uncork (_stream_writable.js:325:7)\n    at JSStreamSocket.doWrite (internal/js_stream_socket.js:162:17)\n    at JSStream.onwrite (internal/js_stream_socket.js:20:57)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"code":"DEPTH_ZERO_SELF_SIGNED_CERT","source":"socket"}}
{"level":"error","message":"Error: self signed certificate\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)\n    at TLSSocket.emit (events.js:210:5)\n    at TLSSocket._finishInit (_tls_wrap.js:794:8)\n    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12)\n    at Socket.ondata (internal/js_stream_socket.js:64:22)\n    at Socket.emit (events.js:210:5)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:290:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:182:23)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"source":"timeout-auth"}}
{"level":"error","message":"Error: Timed out while authenticating with server\n    at Timeout._onTimeout (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:139:17)\n    at listOnTimeout (internal/timers.js:531:17)\n    at processTimers (internal/timers.js:475:7)"}
{"level":"error","message":{"code":"ERR_TLS_INVALID_PROTOCOL_METHOD"}}
{"level":"error","message":{"type":"no","textCode":"ALERT","source":"authentication"}}
{"level":"error","message":"Error: Application-specific password required: https://support.google.com/accounts/answer/185833 (Failure)\n    at Connection._resTagged (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:1502:11)\n    at Parser.<anonymous> (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:194:10)\n    at Parser.emit (events.js:210:5)\n    at Parser._resTagged (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:175:10)\n    at Parser._parse (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:139:16)\n    at Parser._tryread (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:82:15)\n    at TLSSocket.Parser._cbReadable (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:53:12)\n    at TLSSocket.emit (events.js:210:5)\n    at emitReadable_ (_stream_readable.js:575:12)\n    at processTicksAndRejections (internal/process/task_queues.js:79:21)"}
{"level":"error","message":{}}
{"level":"error","message":{"errors":{"tls":{"message":"Path `tls` is required.","name":"ValidatorError","properties":{"message":"Path `tls` is required.","type":"required","path":"tls"},"kind":"required","path":"tls"},"port":{"message":"Path `port` is required.","name":"ValidatorError","properties":{"message":"Path `port` is required.","type":"required","path":"port"},"kind":"required","path":"port"},"host":{"message":"Path `host` is required.","name":"ValidatorError","properties":{"message":"Path `host` is required.","type":"required","path":"host"},"kind":"required","path":"host"},"password":{"message":"Path `password` is required.","name":"ValidatorError","properties":{"message":"Path `password` is required.","type":"required","path":"password"},"kind":"required","path":"password"},"user":{"message":"Path `user` is required.","name":"ValidatorError","properties":{"message":"Path `user` is required.","type":"required","path":"user"},"kind":"required","path":"user"}},"_message":"EmailConfig validation failed","message":"EmailConfig validation failed: tls: Path `tls` is required., port: Path `port` is required., host: Path `host` is required., password: Path `password` is required., user: Path `user` is required.","name":"ValidationError"}}
{"level":"error","message":{"type":"no","textCode":"ALERT","source":"authentication"}}
{"level":"error","message":"Error: Application-specific password required: https://support.google.com/accounts/answer/185833 (Failure)\n    at Connection._resTagged (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:1502:11)\n    at Parser.<anonymous> (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Connection.js:194:10)\n    at Parser.emit (events.js:210:5)\n    at Parser._resTagged (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:175:10)\n    at Parser._parse (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:139:16)\n    at Parser._tryread (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:82:15)\n    at TLSSocket.Parser._cbReadable (C:\\Users\\diogo\\API\\node_modules\\imap\\lib\\Parser.js:53:12)\n    at TLSSocket.emit (events.js:210:5)\n    at emitReadable_ (_stream_readable.js:575:12)\n    at processTicksAndRejections (internal/process/task_queues.js:79:21)"}
